From 944940024ee9c00cc4068329eaf5e7c23996b8ac Mon Sep 17 00:00:00 2001
From: Sunil Kumar Kori <skori@marvell.com>
Date: Mon, 2 Jan 2023 13:26:59 +0530
Subject: [PATCH 105/955] net/cnxk: add threshold validation for RED config

Validation is added to check minimum and maximum RED
threshold values, passed by user.

Fixes: b7d3a0fe71eb ("net/cnxk: support congestion management operations")

Signed-off-by: Sunil Kumar Kori <skori@marvell.com>
Change-Id: I99f79b2d425c97b325e92f56df3efcde3453f992
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/93489
Reviewed-by: Nithin Kumar Dabilpuram <ndabilpuram@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 drivers/net/cnxk/cnxk_ethdev_cman.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/cnxk/cnxk_ethdev_cman.c b/drivers/net/cnxk/cnxk_ethdev_cman.c
index d5e647c64da12..a7ccdfb756f30 100644
--- a/drivers/net/cnxk/cnxk_ethdev_cman.c
+++ b/drivers/net/cnxk/cnxk_ethdev_cman.c
@@ -68,6 +68,11 @@ nix_cman_config_validate(struct rte_eth_dev *eth_dev, const struct rte_eth_cman_
 		return -EINVAL;
 	}
 
+	if (config->mode_param.red.min_th > config->mode_param.red.max_th) {
+		plt_err("RED minimum threshold must be less or equal to maximum threshold");
+		return -EINVAL;
+	}
+
 	return 0;
 }
 
-- 
2.25.1

